bitcoin/src/consensus/params.rs

```diff
@@ -121,6 +136,7 @@ impl From<Network> for Params {
     fn from(value: Network) -> Self { Self::new(value) }
 }
 
+
```

Comment:
Seems unneeded.

Reply:
Can we have a policy that any comment thread that starts with a "nit: ..." message can be resolved by the merger? (Slightly off topic, I just thought of it yesterday and here seems like a good place to mention it.(

Reply:
That sounds fine to me, though should it be "resolved" if the nit isn't actually addressed?

Reply:
I thought "nit" meant "fix if you  are force pushing anyways but if this has two acks then do it later or not at all". I didn't want to click resolve though in case at some later time I had to force push, so I can be reminded of the open nits still - although I usually just fix nits anyways, for time sensitive PRs this matters.

Reply:
Another option would be that nits can be handled by the OP using some other method (personal todo list) and marking resolved for them just means "I saw and read this, thanks"

Reply:
Personally I'm leaning toward "we shouldn't require conversation resolution". I feel like if somebody opens a nit, the nit doesn't get fixed, and we merge the PR anyway, that's fine....but then the PR should show that the nit was raised and was never addressed. If you click "resolve" then it gets hidden and it becomes super annoying to try to find the nit.

Our "one ACK required" setting already prevents merging if anybody has left a review where they marked "changes requested". Maybe we should just be more disciplined about using "changes requested" for things that we actually want to be addressed.

Reply:
You seem to be overcomplicating things. :) If I ACKed anyway it implies this is fine and just found something that could be improved if wanted. In this case simply replying "I don't want to fix it" and clicking resolve is enough.

IOW I hereby give anyone permission to resolve my comments within the review in which I ACKed a PR with either of these actions:

* PR author fixes the issue
* PR author opens a new issue for it (or anyone else who believes the PR is urgent)
* anyone writes a "not worth bothering" response

> Personally I'm leaning toward "we shouldn't require conversation resolution".

I think ideally the rule would be "resolve all conversation except if the reviewer ACKed". AFAIK this is not automatable with GitHub.

> Maybe we should just be more disciplined about using "changes requested"

The issue is that sometimes I'm not confident that something is wrong and want a second opinion. In those cases I don't click changes requested but I want my comment to be addressed - I don't want it to be missed.

Reply:
> In this case simply replying "I don't want to fix it" and clicking resolve is enough.

In this case that's fine, it's just a whitespace that has no effect on the code. But what often happens is that a "nit" turns out, years later, to be the only conversation we had about a design change that seemed trivial at the time but later turned out to be consequential.

And then when I go through the PR looking for whether this issue was brought up, instead of having a clear "nit: this makes X do Y when Z happens, but it's probably fine" with no response, I have 2 dozen "conversation resolved" that I have to manually click on to open up, and probably every single one is some irrelevant thing about a version of the PR that wasn't even merged. But maaybe it turns out that one is relevant.

Ok, if we have a policy that you have to explicitly say "not worth bothering", that will save me the time of reverse-engineering whether an apparently-relevant conversation actually even applies to the version of the code that was merged. But it's still a very bad user experience.

TLDR My issue is that the "resolve" button basically deletes the conversation for practical purposes.

Reply:
Hmm, yeah, that's annoying. But is it really worse than forgetting to address a review comment? I suspect digging into history is less frequent than forgetting.

Reply:
As a recent example, from the last 5 minutes, see https://github.com/rust-bitcoin/rust-bitcoin/pull/2530#pullrequestreview-1922126999 -- this turned up an issue with a 2-year-old PR with 250+ comments. There are plenty of still-visible comments. The issue under question is not one of them. This is useful information.

Reply:
Well that example has a bunch of junk there as well that would get hidden if we clicked resolved.

Maybe we could require that whoever merges and clicks "Resolve" needs to record that they did so with justification in a comment. The justification can be as simple as a new issue number. Formatting changes are exempt from this rule.

This would improve the situation since the unresolved things would be grouped and easily found.

Reply:
>The justification can be as simple as a new issue number.

Hmmm, *this* I can get behind. You're allowed to click "resolve" on unresolved stuff but you need to file an issue.

>Formatting changes are exempt from this rule.

For sure.

Reply:
It's still not perfect; when I use `git blame` I wind up on the PR rather than having direct access to filed issues. But I think when you open an issue from a comment, it shows up in the PR as "apoelstra referenced this in issue #whatever" and that notification does not get hidden.

Reply:
I meant writing justification in a separate comment in the PR - not a discussion. So you'd always see it as the last comment. But yes "referenced by" is also helpful.

Reply:
Ah, gotcha! Ok, ack, let's try that.

Reply:
Ok, I think we've resolved this side-discussion about resolving discussions.

Meanwhile the original line of code under consideration was a style nit :P. So @Kixunil can we hit "resolve" on this and merge?

Reply:
Of course.